Sources/ManageBoards.php

```diff
@@ -394,13 +395,14 @@ function EditBoard()
 
 	// id_board must be a number....
 	$_REQUEST['boardid'] = isset($_REQUEST['boardid']) ? (int) $_REQUEST['boardid'] : 0;
+
 	if (!isset($boards[$_REQUEST['boardid']]))
 	{
 		$_REQUEST['boardid'] = 0;
 		$_REQUEST['sa'] = 'newboard';
 	}
 
-	if ($_REQUEST['sa'] == 'newboard')
+	if ('newboard' === $_REQUEST['sa'])
```

Comment:
why you add this "new" style every where?

Reply:
They are Yoda conditions: https://en.wikipedia.org/wiki/Yoda_conditions

They help to prevent accidental variable assigning. We should also start to use real comparisons (value and type) whenever possible

Reply:
Yeah i mean the yoda conditions,
from my pov this base issue of "=" didn't exists in php and
the worst reading is a real thing.

Reply:
PHP has always been a loosely typed language, don't really get what you mean by "didn't exists in php". If you mean SMF, yes, we have always used "loose" instead of "strict", that doesn't mean we should continue to use it on things we should be using strict compares.

As for yoda conditions, you are just changing the order, putting the constant first makes you realize it is comparing and not assigning. Helps you from stop taking things for granted too.

Reply:
I don't understand why you mix upo different thing,
the type checks is fine, nothing wrong here.

Only the yoda style is unneed because singel = had only the task to assign value not to check them.

Reply:
Putting the constant first helps you from accidentally assigning a variable, take this example:

$var == 'constant';

Imaging you accidentally remove a = 

$var = 'constant';

Now $var means 'constant'; and it will always return true.  This happens way more frequent that you think, specially on large codebases touched by several people.

Reply:
but we talk about = vs === and here remove accidentally a = change not the equality check.

Reply:
Yes.... thats exactly what this change does, besides doing the yoda conditions, using strict comparisons helps out too.

In resume, yoda comparisons prevents accidental assigning and helps out figure it out we are doing a compare and not an assign.

We should use strict comparisons whenever they are needed. Such as this case.

Those two things (yoda and strict) are separated entities and will not always be used together like on this example. There might be cases where only one will be applied.

And I believe I already spent a lot more comments than needed explaining this change, if there are further questions these links will help out:

https://en.wikipedia.org/wiki/Yoda_conditions
https://www.php.net/manual/en/types.comparisons.php